### PR TITLE
Member Detail UI

### LINKED ISSUE #43: Implement Member Detail Page UI
## Description
As a user, I need to see the complete details about a member so I could know more about the community members and send relation requests.

## Mocks
[INSERT RELEVANT PNG FILE]

### Update [Required]
- [ ] Develop member detail screen
- [ ] The screen shall show all available details about the member tapped in list screen
- [ ] Actual data should be used, taken from members list screen
- [ ] The screen should have a button for sending request
- [ ] Send request screen should be included

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
[6] hours

### PR DESCRIPTION
**Pull request by yugantarjain**
### Description
Adds the screen for Member Detail and Send Request screen. 

Additionally:
1. Develops tab bar with proper icons and naming
2. Sets global tint color to systemindigo
3. Adds debug comments in a few places where code has to be changed later. This is done to ease testing and also due to some swiftui bugs.

Fixes #43 

### Mocks:
Light | Dark
----- | -----
Screen Shot 2020-06-09 at 3 13 32 PM | Screen Shot 2020-06-09 at 3 13 36 PM
Screen Shot 2020-06-09 at 5 28 11 PM | Screen Shot 2020-06-09 at 5 28 04 PM

### Type of Change:
- Code
- User Interface
- Outreach

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Tested on simulator. Travis checked locally using swiftlint.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: mentorship ios/Members/SendRequest.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                //heading
+                Section(header: Text("To \(name)").font(.title).fontWeight(.heavy)) {
+                    EmptyView()
+                }
+                
+                //settings
+                Section {
+                    Picker(selection: $pickerSelection, label: Text("My Role")) {
+                        Text("Mentee").tag(1)
+                        Text("Mentor").tag(2)
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localize "Mentor" "Mentee" "My Role"

File: mentorship ios/Members/SendRequest.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                    }
+                    
+                    DatePicker(selection: $endDate, displayedComponents: .date) {
+                        Text("End Date")
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localize "End Date"

File: mentorship ios/Members/SendRequest.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    }
+                    
+                    TextField("Notes", text: $notesText)
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localize "Notes"

File: mentorship ios/Members/SendRequest.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                }
+                .padding(.vertical, DesignConstants.Padding.listCellFrameExpansion)
+                
+                //send button
+                Section {
+                    Button(action: {}) {
+                        Text("Send")
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localize "Send"

File: mentorship ios/Members/SendRequest.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+                    }
+                }
+            }
+            .navigationBarTitle("Relation Request")
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localize all text " Relation request"

File: mentorship ios/Members/SendRequest.swift [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            .navigationBarItems(leading: Button.init("Cancel", action: {
```
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: Localise "Cancel"

File: mentorship ios/Models/MembersModel.swift [hunk 7]
```diff
@@ -33,10 +34,9 @@ final class MembersModel: ObservableObject {
                 }
             }, receiveValue: { value in
                 self.membersResponseData = value
-                print(value)
             })
     }
-    
+
     func availabilityString(canBeMentee: Bool, canBeMentor: Bool) -> String {
```
Comment (ApheleiaS) [inline mentorship ios/Models/MembersModel.swift:38]: Localize

File: mentorship ios/Managers/NetworkManager.swift [hunk 8]
```diff
@@ -11,18 +11,20 @@ import Combine
 
 struct NetworkManager {
     static var responseCode: Int = 0
-    
-    static func callAPI<T: Decodable>(urlString: String, httpMethod: String, uploadData: Data, token: String = "") -> AnyPublisher<T, Error> {
+
+    // swiftlint:disable:next all
+    static func callAPI<T: Decodable>(urlString: String, httpMethod: String, uploadData: Data, token: String = "", cachePolicy: NSURLRequest.CachePolicy = .useProtocolCachePolicy) -> AnyPublisher<T, Error> {
```
Comment (ApheleiaS) [inline mentorship ios/Managers/NetworkManager.swift]: Break up function declaration into multiline and remove swiftlint ignore

File: mentorship ios/Members/MemberDetail.swift [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        .sheet(isPresented: $showSendRequestSheet) {
+            SendRequest(name: self.member.name ?? "-")
+        }
+    }
+}
+
+struct MemberDetail_Previews: PreviewProvider {
+    static var previews: some View {
+        // swiftlint:disable:next all
+        MemberDetail(member: MembersModel.MembersResponseData.init(id: 1, username: "username", name: "yugantar", bio: "student", location: "earth", occupation: "student", organization: "", interests: "astronomy", skills: "ios, swift, c++", slackUsername: "", needMentoring: true, availableToMentor: true, isAvailable: true))
```
Comment (ApheleiaS) [inline mentorship ios/Members/MemberDetail.swift]: Break up function call into multiline and remove swiftlint ignore

Comment (ApheleiaS) [reply mentorship ios/Members/MemberDetail.swift]: https://google.github.io/swift/#function-declarations
or Section 1.10 here https://github.com/linkedin/swift-style-guide/blob/master/README.md

File: mentorship ios/Managers/NetworkManager.swift (hunk 8 above)
Comment (ApheleiaS) [reply mentorship ios/Managers/NetworkManager.swift]: Read up on Line Wrapping here https://google.github.io/swift/#function-declarations
or https://github.com/linkedin/swift-style-guide/blob/master/README.md Section 1.9

File: mentorship ios/Managers/KeyboardManager.swift [hunk 10]
```diff
@@ -0,0 +1,27 @@
+////
+////  KeyboardManager.swift
+////  mentorship ios
+////
+////  Created by Yugantar Jain on 09/06/20.
+////  Copyright © 2020 Yugantar Jain. All rights reserved.
```
Comment (ApheleiaS) [inline mentorship ios/Managers/KeyboardManager.swift:6]: Remove copyright

File: mentorship ios/Members/Cells/MemberDetailCell.swift [hunk 11]
```diff
@@ -0,0 +1,38 @@
+//
+//  MemberDetailCell.swift
+//  mentorship ios
+//
+//  Created by Yugantar Jain on 09/06/20.
+//  Copyright © 2020 Yugantar Jain. All rights reserved.
```
Comment (ApheleiaS) [inline mentorship ios/Members/Cells/MemberDetailCell.swift:6]: Remove Copyright

File: mentorship ios/Members/MemberDetail.swift [hunk 12]
```diff
@@ -0,0 +1,48 @@
+//
+//  MemberDetail.swift
+//  mentorship ios
+//
+//  Created by Yugantar Jain on 08/06/20.
+//  Copyright © 2020 Yugantar Jain. All rights reserved.
```
Comment (ApheleiaS) [inline mentorship ios/Members/MemberDetail.swift:6]: Remove Copyright

**Comment by sunjunkie** [comment]
Change "Available to be a Mentor and Mentee both" to "Available to be a Mentor or Mentee"

File: mentorship ios/TabBar.swift [hunk 13]
```diff
@@ -10,22 +10,22 @@ import SwiftUI
...
+
     var body: some View {
         TabView(selection: $selection) {
             Home()
                 .tabItem {
                     VStack {
-                        Image("first")
-                        Text("First")
+                        Image(systemName: "house.fill")
+                        Text("Home")
```
Comment (annabauza) [inline mentorship ios/TabBar.swift:18]: has key been added?

File: mentorship ios/Managers/KeyboardManager.swift [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+////
+//
+//import Foundation
+//import Combine
+//import SwiftUI
+//
+//final class KeyboardManager: ObservableObject {
```
Comment (annabauza) [inline mentorship ios/Managers/KeyboardManager.swift:13]: remove this file

File: mentorship ios/Members/MemberDetail.swift (hunk 12 above)
Comment (sunjunkie) [reply mentorship ios/Members/MemberDetail.swift:6]: We need to customize the xcode headers. I will submit an issue for that.

File: mentorship ios/Managers/KeyboardManager.swift (hunk 14 above)
Comment (yugantarjain) [reply mentorship ios/Managers/KeyboardManager.swift:13]: This file will surely be used later. So I'd prefer to keep it.

File: mentorship ios/Members/MemberDetail.swift (hunk 12 above)
Comment (vatsalkul) [reply mentorship ios/Members/MemberDetail.swift:6]: solved in #59

File: mentorship ios/Members/Cells/MemberDetailCell.swift (hunk 11 above)
Comment (vatsalkul) [reply mentorship ios/Members/Cells/MemberDetailCell.swift:6]: solved in pull request 59

File: mentorship ios/Managers/KeyboardManager.swift (hunk 10 above)
Comment (vatsalkul) [reply mentorship ios/Managers/KeyboardManager.swift:6]: solved in pull request 59

**Review by annabauza — APPROVED** [review]

**Pull request merged**